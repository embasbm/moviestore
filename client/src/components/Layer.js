import React, { Component } from 'react';
import logo from '../assets/logo.svg';
import '../styles/Layer.css';
import MoviesContainer from './MoviesContainer';

class Layer extends Component {
  render() {
    return (
      <div className="movie-app">
        <header className="movie-app-header">
          <img src={logo} className="movie-app-logo" alt="logo" />
          <h1 className="movie-app-title">Movie DB</h1>
        </header>
        <MoviesContainer />
      </div>
    );
  }
}

export default Layer;
