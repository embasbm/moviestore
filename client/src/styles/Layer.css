.movie-app {
  text-align: center;
}

.movie-app-logo {
  animation: movie-app-logo-spin infinite 20s linear;
  height: 80px;
}

.movie-app-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.movie-app-title {
  font-size: 1.5em;
}

.movie-app-intro {
  font-size: large;
}

@keyframes movie-app-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
