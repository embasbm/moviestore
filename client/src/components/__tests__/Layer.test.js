import React from 'react';
import axios from "axios";
const ReactTestRenderer = require('react-test-renderer');
import Layer from '../Layer';

jest.mock('axios');

describe('Layer', () => {
  const resp = {
    data: []
  };
  axios.get.mockImplementation(() => Promise.resolve(resp));

  it('Should compare Layer component with a snapshot', () => {
    const component = ReactTestRenderer.create(<Layer />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  })
});
