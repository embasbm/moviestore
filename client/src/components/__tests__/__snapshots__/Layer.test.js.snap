// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layer Should compare Layer component with a snapshot 1`] = `
<div
  className="movie-app"
>
  <header
    className="movie-app-header"
  >
    <img
      alt="logo"
      className="movie-app-logo"
      src="logo.svg"
    />
    <h1
      className="movie-app-title"
    >
      Movie DB
    </h1>
  </header>
  <div
    className="movies-container"
  />
</div>
`;
